### DISCUSSION TITLE
Adding new property "learningOutcomeIds' to questionSet Schema

### DISCUSSION THREAD

**Discussion by vaivk369** [Sunbird-inQuiry/Community#91] [General Discussions]
Hi @krgauraw 

We are trying to add new property `learningOutcomeIds` to questionset. 
Steps I followed:
1. Updated [kp_schemas/questionset/1.0/schema.json](https://github.com/Sunbird-coKreat/creation-portal/pull/2222/files#diff-83add3b3dd1843123651b27ffde0b97a32a8055e32cbe50c3caf3f99490ca751) to have `learningOutcomeIds` in the schema file
2. Executed https://10.20.0.14/jenkins/job/Deploy/job/DockDev/job/Kubernetes/job/UploadSchemas/ - SUCCESS
3. Verifies schema is updated- [https://dockstorage.blob.core.windows.net/content-service/schemas/questionset/1.0/schema.json](https://dockstorage.blob.core.windows.net/content-service/schemas/questionset/1.0/schema.json) 

But still I am getting `Invalid Props are : [learningOutcomeIds]` error while creating questionset with API CURL as:

`curl 'https://dock.sunbirded.org/action/questionset/v1/create' \
  -H 'Accept: application/json' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: connect.sid=s%3ATroiMcXla6MI4BNy6O51LXuLUh2ZcxIT.0vuG36szLApXkUJ01q%2FPoYMlwuPtFNfseWLiqBRqWS8' \
  --data-raw '{"request":{"questionset":{"name":"Untitled","code":"a563f3fb-35e8-4ddf-a9f4-cce7ed01ecbd","mimeType":"application/vnd.sunbird.questionset","createdBy":"366a95a0-35d5-4408-8c37-8d60f8fab0da","primaryCategory":"Practice Question Set","creator":"cokreat","author":"cokreat","programId":"7f8187c0-0c01-11ee-9d2c-c1818048f076","channel":"01379157488516300825","framework":"vdn_k-12","board":"Pune","medium":[],"gradeLevel":[],"subject":[],"boardIds":[],"mediumIds":[],"gradeLevelIds":[],"subjectIds":[],"topic":[],"learningOutcome":[],"learningOutcomeIds":[]}}}' \
  --compressed`

Response: 
{
    "id": "api.questionset.create",
    "ver": "3.0",
    "ts": "2023-06-28T07:05:59ZZ",
    "params": {
        "resmsgid": "840d4de3-9916-48c6-82cf-18604c38f5e0",
        "msgid": null,
        "err": "CLIENT_ERROR",
        "status": "failed",
        "errmsg": "Invalid request"
    },
    "responseCode": "CLIENT_ERROR",
    "result": {
        "messages": [
            "Invalid Props are : [learningOutcomeIds]"
        ]
    }
}

cc: @vinukumar-vs

**Comment by vaivk369** [comment]
@krgauraw did you get a chance to look into this?. It would be great if you could spare some time and help.

**Reply by krgauraw** [reply]
@vaivk369 Please restart the assessment-service

**Reply by vaivk369** [reply]
I have rebuilt the assessment service and it seems working fine. Thank you @krgauraw

**Reply by vaivk369** [reply]
Steps followed to resolve the issue:

1. Updated [kp_schemas/questionset/1.0/schema.json](https://github.com/Sunbird-coKreat/creation-portal/pull/2222/files#diff-83add3b3dd1843123651b27ffde0b97a32a8055e32cbe50c3caf3f99490ca751) to have learningOutcomeIds in the schema file
2. Executed https://10.20.0.14/jenkins/job/Deploy/job/DockDev/job/Kubernetes/job/UploadSchemas/ - SUCCESS
3. Verified schema is updated- https://dockstorage.blob.core.windows.net/content-service/schemas/questionset/1.0/schema.json
4. Rebuilt the assessment service: https://10.20.0.14/jenkins/job/Build/job/DockCore/job/Assessment/88/
5. Redeployed assessment service:  https://10.20.0.14/jenkins/job/Deploy/job/DockDev/job/Kubernetes/job/Assessment/
